Mathlib/LinearAlgebra/TensorAlgebra/Hopf.lean

```diff
@@ -0,0 +1,210 @@
+/-
+Copyright (c) 2025 Nikolas Tapia. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Nikolas Tapia
+-/
+module
+
+public import Mathlib.LinearAlgebra.TensorAlgebra.Basic
+public import Mathlib.RingTheory.Coalgebra.Convolution
+public import Mathlib.RingTheory.HopfAlgebra.Basic
+
+/-!
+# Hopf algebra structure on `TensorAlgebra R M`
+
+In this file we implement the natura Hopf algebra structure on `TensorAlgebra R M`.
+The comultiplication is the unique algebra map satisfying `comul ((ι R) x) = (ι R) x ⊗ 1 + 1 ⊗ (ι R)
+x` for all `x : M`.
+`algebraMapInv` acts as the counit, and the antipode is the unique algebra map `antipode :
+TensorAlgebra R M → (TensorAlgebra R M)ᵐᵒᵖ` induced by `fun x => op -(ι R) x`.
+-/
+@[expose] public section
+
+namespace TensorAlgebra
+open scoped TensorProduct RingTheory.LinearMap Coalgebra
+open LinearMap TensorProduct
+
+variable (R : Type*) [CommRing R] {M : Type*} [AddCommMonoid M] [Module R M]
```

Comment:
```suggestion
variable (R : Type*) [CommRing R] {M : Type*} [AddCommGroup M] [Module R M]
```

Since `R` is a ring, any module over it is automatically an AddCommGroup already, so by convention we write it this way.